tools/bumper/parser_test.go

```diff
@@ -0,0 +1,198 @@
+				createTempFileWithContent(emptyYaml)
+			})
+			It("should return an error", func() {
+				_, err := parseComponentsYaml(tmpfile.Name())
+				Expect(err).To(HaveOccurred(), "Should Fail to parse file")
+			})
+		})
+		Context("when yaml file is corrupted", func() {
+			BeforeEach(func() {
+				By("Creating a temporary file with corrupted content")
+				createTempFileWithContent(corruptedDummyYaml)
+			})
+			It("should return an error because the unmarshaling failed", func() {
+				By(fmt.Sprintf("Parsing content of temporary file %s", tmpfile.Name()))
+				_, err := parseComponentsYaml(tmpfile.Name())
+				Expect(err).To(HaveOccurred(), "Should Fail to parse file")
+			})
+		})
+		Context("when yaml file is valid", func() {
+			BeforeEach(func() {
+				By("Creating a temporary file with empty content")
+				createTempFileWithContent(componentsDummyYaml)
+			})
+			It("should correctly parse params of all components", func() {
+				By(fmt.Sprintf("Parsing content of temporary file %s", tmpfile.Name()))
+				componentsConfig, err := parseComponentsYaml(tmpfile.Name())
+				Expect(err).ToNot(HaveOccurred(), "Should Succeed to parse file")
+
+				By("Checking map is not empty")
+				Expect(componentsConfig.Components).ToNot(BeEmpty(), "Components map Should not be empty")
+				By("Checking map output is as expected")
+				Expect(componentsConfig).To(Equal(&expectedValidComponentsConfig), "componentsConfig should be identical to expected")
+			})
+		})
+	})
+
+	Describe("UpdateComponentsYaml function", func() {
+		Context("when yaml file is valid", func() {
+			var componentsConfig *componentsConfig
+			var err error
+
+			BeforeEach(func() {
+				By("Creating a temporary file with empty content")
+				createTempFileWithContent(componentsDummyYaml)
+
+				By("Parsing the yaml")
+				componentsConfig, err = parseComponentsYaml(tmpfile.Name())
+				Expect(err).ToNot(HaveOccurred(), "Should Succeed to parse file")
+			})
+
+			It("should correctly update existing component in file", func() {
+				By("Changing some params in existing component the yaml")
+				newComponent := component{Url: "https://github.com/qinqon/kube-admission-webhook", Commit: "f8f7795d05781cdaf869c301e4736adc29a0de28", Metadata: "v0.11.0"}
+				componentsConfig.Components["MacvtapCni"] = newComponent
+
+				err := updateComponentsYaml(tmpfile.Name(), componentsConfig)
+				Expect(err).ToNot(HaveOccurred(), "Should Succeed to updating the file")
+
+				By("Checking params updated in the yaml")
+				newcomponentsConfig, err := parseComponentsYaml(tmpfile.Name())
+
+				By("Checking MacvtapCni params are valid")
+				Expect(newcomponentsConfig.Components["MacvtapCni"].Url).To(Equal("https://github.com/qinqon/kube-admission-webhook"))
+				Expect(newcomponentsConfig.Components["MacvtapCni"].Commit).To(Equal("f8f7795d05781cdaf869c301e4736adc29a0de28"))
+				Expect(newcomponentsConfig.Components["MacvtapCni"].Metadata).To(Equal("v0.11.0"))
```

Comment:
Create a component and compare it to reduce the number of Expects.

Reply:
Done